packages/components/src/tools-panel/types.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { ReactNode } from 'react';
+
+type ResetAll = ( filters?: any[] ) => void;
+type PanelId = string;
+type ResetAllFilter = () => void;
+type Label = string;
+
+export interface ToolsPanelProps {
+	panelId: PanelId;
+	label: Label;
+	header: string;
+	children: ReactNode;
+	resetAll: ResetAll;
+	className?: string;
+}
+
+export interface ToolsPanelHeaderProps {
+	label: Label;
+	resetAll: ResetAll;
+	toggleItem: ( label: string ) => void;
+}
+
+export interface ToolsPanelItem {
+	hasValue: () => boolean;
+	isShownByDefault: boolean;
+	label: Label;
+	resetAllFilter: ResetAllFilter;
+	panelId: PanelId;
+}
+
+export interface ToolsPanelItemProps extends ToolsPanelItem {
+	panelId: PanelId;
+	children?: ReactNode;
+	onDeselect?: () => void;
+	onSelect?: () => void;
+	className?: string;
```

Comment:
`className` shouldn't be needed here, since `ToolsPanelItem` uses `WordPressComponentProps`